File: 121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
  let profit = 0;
  let highest = 0;
  let lowest = 0;

  for (let i = 0; i < prices.length; i++) {
    if (prices[i] < prices[lowest]) {
      lowest = i;
      highest = i;
    }

    if (prices[i] > prices[highest]) {
      highest = i;
      let diff = prices[highest] - prices[lowest];

      profit = Math.max(diff, profit);
    }
  }

  return profit;
}

console.log(maxProfit([7, 1, 5, 3, 6, 4]));

File: 121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        output = 0
        lowest = 0
        highest = 0

        for n in range(len(prices)):
            if (prices[n] < prices[lowest]):
                lowest = n
                highest = n
            if (prices[n] > prices[highest] and n > lowest):
                highest = n
                new_output = prices[highest] - prices[lowest]
                if (new_output > output):
                    output = new_output
        

        return output